File: haskell/006/Main.hs
{-
    ProjectEuler 006
    Haskell / Main.hs
    copyright (c) 2015 Susisu

    GHC 7.8.3
    `ghc Main.hs -o bin/main`
-}

module Main where

main :: IO ()
main = print $ calc 1 100 0 0

calc :: Int -> Int -> Int -> Int -> Int
calc n max s t =
    if n <= max then
        calc (n + 1) max (s + n) (t + n ^ 2)
    else
        s ^ 2 - t

File: go/006/main.go
/*
 * ProjectEuler 006
 * Go / main.go
 * copyright (c) 2015 Susisu
 *
 * Go 1.4.1
 * `go build -o bin/main main.go`
 */


package main

import "fmt"

func main() {
    s := 0
    t := 0
    for n := 1; n <= 100; n++ {
        s += n
        t += n * n
    }
    fmt.Printf("%d\n", s * s - t)
}
